common/scala/src/main/scala/org/apache/openwhisk/common/TransactionId.scala

```diff
@@ -172,13 +175,22 @@ case class TransactionId private (meta: TransactionMetadata) extends AnyVal {
   def deltaToMarker(startMarker: StartMarker, endTime: Instant = Instant.now(Clock.systemUTC)) =
     Duration.between(startMarker.start, endTime).toMillis
 
+  def hasParent = meta.parent.isDefined
+
   /**
    * Formats log message to include marker.
    *
    * @param message: The log message without the marker
    * @param marker: The marker to add to the message
    */
   private def createMessageWithMarker(message: String, marker: LogMarker): String = s"$message $marker"
+
+  /**
+   * Find root transaction metadata
+   */
+  @tailrec
+  private def findRoot(meta: TransactionMetadata): TransactionMetadata =
+    if (meta.parent.isDefined) findRoot(meta.parent.get) else meta
```

Comment:
```suggestion
    meta.parent match {
       case Some(parent) => findRoot(parent)
       case _ => meta
```

Reply:
avoid `.get` even though it's safe, also clearer to me

Reply:
@rabbah Thank you for your.sample code, I've modified the code.